LanguageFeatures/Inline-classes/syntax_A02_t08.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A rule for <inlineClassDeclaration> is added to the grammar,
+/// along with some rules for elements used in inline class declarations:
+///
+/// <inlineClassDeclaration> ::=
+///   'final'? 'inline' 'class' <typeIdentifier> <typeParameters>? <interfaces>?
+///   '{'
+///     (<metadata> <inlineMemberDeclaration>)*
+///   '}'
+///
+/// <inlineMemberDeclaration> ::= <classMemberDefinition>
+///
+/// @description Checks that an inline class can be declared as `final` and can
+/// be extended and implemented by `final inline` classes in the same library
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+final inline class FIC {
+  final int id ;
+  const FIC(this.id);
+}
+
+final inline class FIC2 extends FIC {
```

Comment:
This is also a compile-time error: An inline class can not extend anything (we've had discussions about introducing that as a feature, but it is not part of the currently accepted proposal).

The point is that an inline class never has a subclass relationship to another inline class: The "superclass" has a single field, but the "subclass" doesn't inherit that field, it has to declare a new one. So the relationship is a subtype relationship, as in `implements`.

This is also a good match for the understanding of inline class methods as similar in nature to extension methods: `inline class B implements A1, .. Ak {}` is able to "inherit" members from `A1`, from `A2` etc. because that's how an extension method `on A1` or `on A2` etc. works. Also, if `A1` has `bool get m => true;` and `A2` has `void m() {}`, there is no error as long as `B` declares its own `m`. There is no requirement that `B.m` should be a correct override of `A1.m` or `A2.m`. Again, just like extension methods.